cloudsync/event.py

```diff
@@ -132,13 +133,14 @@ def _reconnect_if_needed(self):
             else:
                 log.info("reconnect to %s", self.provider.name)
                 self.provider.reconnect()
-            self._validate_root()
 
     def do(self):
         try:
             self._reconnect_if_needed()
-            self._do_unsafe()
+            if self._validate_root():
```

Comment:
we should do this once per connection i think.... not every time we get events.

Reply:
sure, not a bad idea to re-validate on reconnect.

was also thinking we can detect root problems in event manager while connected (if root folder is renamed or deleted we should get an event for it, and can raise an error / notify / whatever)

Reply:
added root validation on connect in provider.py